fairseq/tasks/sequence_tagging.py

```diff
@@ -0,0 +1,242 @@
+                            help='add prev_output_tokens to sample, used for encoder-decoder arch')
+
+    def __init__(self, args, data_dictionary, label_dictionary):
+        super().__init__(args)
+        self.dictionary = data_dictionary
+        self._label_dictionary = label_dictionary
+        if not hasattr(args, 'max_positions'):
+            self._max_positions = (
+                args.max_source_positions,
+                args.max_target_positions,
+            )
+        else:
+            self._max_positions = args.max_positions
+        args.tokens_per_sample = self._max_positions
+
+    @classmethod
+    def load_dictionary(cls, args, filename, source=True):
+        """Load the dictionary from the filename
+
+        Args:
+            filename (str): the filename
+        """
+        dictionary = Dictionary.load(filename)
+        dictionary.add_symbol('<mask>')
+        return dictionary
+
+    @classmethod
+    def setup_task(cls, args, **kwargs):
+        assert args.num_classes > 0, 'Must set --num-classes'
+
+        # load data dictionary
+        data_dict = cls.load_dictionary(
+            args,
+            os.path.join(args.data, 'input0', 'dict.txt'),
+            source=True,
+        )
+        logger.info('[input] dictionary: {} types'.format(len(data_dict)))
+
+        label_dict = None
+        if not args.regression_target:
+            # load label dictionary
+            label_dict = cls.load_dictionary(
+                args,
+                os.path.join(args.data, 'label', 'dict.txt'),
+                source=False,
+            )
+            logger.info('[label] dictionary: {} types'.format(len(label_dict)))
+        else:
+            label_dict = data_dict
+        return SequenceTaggingTask(args, data_dict, label_dict)
+
+    def load_dataset(self, split, combine=False, **kwargs):
+        """Load a given dataset split (e.g., train, valid, test)."""
+        def get_path(type, split):
+            return os.path.join(self.args.data, type, split)
+
+        def make_dataset(type, dictionary):
+            split_path = get_path(type, split)
+
+            dataset = data_utils.load_indexed_dataset(
+                split_path,
+                dictionary,
+                self.args.dataset_impl,
+                combine=combine,
+            )
+            return dataset
+
+        input0 = make_dataset('input0', self.source_dictionary)
+        assert input0 is not None, 'could not find dataset: {}'.format(get_path(type, split))
+        input1 = make_dataset('input1', self.source_dictionary)
+
+        if self.args.init_token is not None:
+            input0 = PrependTokenDataset(input0, self.args.init_token)
+
+        if input1 is None:
+            src_tokens = input0
+        else:
+            if self.args.separator_token is not None:
+                input1 = PrependTokenDataset(input1, self.args.separator_token)
```

Comment:
Do we ever have two inputs for this kind of task?

Reply:
Good point, I removed this functionality